src/github.com/matrix-org/rageshake/submit.go

```diff
@@ -93,12 +95,20 @@ func (s *submitServer) saveReport(ctx context.Context, p payload) error {
 	//  "bugreport-20170115-112233-N.log.gz" => oldest log
 	t := time.Now().UTC()
 	prefix := t.Format("2006-01-02/150405")
-	summary := fmt.Sprintf(
-		"%s\n\nNumber of logs: %d\nVersion: %s\nUser-Agent: %s\n", p.Text, len(p.Logs), p.Version, p.UserAgent,
+
+	var summaryBuf bytes.Buffer
+	fmt.Fprintf(
+		&summaryBuf,
+		"%s\n\nNumber of logs: %d\nApplication: %s\nVersion: %s\nUser-Agent: %s\n",
```

Comment:
bytes.Buffer is a writer, so why can't I write to it? Why bother going through a string for the formatting?

As for using `buf.String()`: I want to end up with the bytes (to stick into `gzipAndSave`), so again, why bother turning it into an intermediate string?